cursorless-talon/src/targets/destination.py

```diff
@@ -0,0 +1,70 @@
+from typing import Union
+
+from talon import Context, Module, actions
+
+from .target_types import ListDestination, PrimitiveDestination
+
+mod = Module()
+
+mod.list(
+    "cursorless_insertion_mode_before_after",
+    desc="Cursorless insertion mode before/after",
+)
+mod.list("cursorless_insertion_mode_to", desc="Cursorless insertion mode to")
+mod.tag(
+    "cursorless_disable_legacy_destination",
+    desc="Disabled the Cursorless legacy destination(to after) support",
+)
+
+ctx = Context()
+ctx.matches = r"""
+tag: user.cursorless_disable_legacy_destination
+"""
+
+
+# DEPRECATED @ 2023-08-01
+@mod.capture(
+    rule="([{user.cursorless_insertion_mode_to}] {user.cursorless_insertion_mode_before_after}) | {user.cursorless_insertion_mode_to}"
+)
+def cursorless_insertion_mode(m) -> str:
+    try:
+        before_after = m.cursorless_insertion_mode_before_after
+        if hasattr(m, "cursorless_insertion_mode_to"):
+            words = m._unmapped
+            actions.app.notify(
+                f"'{' '.join(words)}' is deprecated. Please just say '{words[-1]}'"
```

Comment:
Not sure whether it's worth fixing, but this doesn't correctly handle the case where they have more than one word in their spoken form for `"before"` or `"after"`, right?

Reply:
Right but eg if they have `"prior to"` as their spoken form for `before`, then if they say `"to prior to"`, it will say ``to prior to` is deprecated. Please just say 'to'`, correct? I guess we can live with that; I have to imagine remapping `"after"` to a two-word modifier is exceedingly rare, if anyone has done it at all

Reply:
You are probably right. Should be very rare